win32ss/gdi/gdi32/misc/misc.c

```diff
@@ -745,7 +745,7 @@ GdiFixUpHandle(HGDIOBJ hGdiObj)
     Entry = GdiHandleTable + GDI_HANDLE_GET_INDEX(hGdiObj);
 
     /* Rebuild handle for Object */
-    return hGdiObj = (HGDIOBJ)(((LONG_PTR)(hGdiObj)) | (Entry->Type << GDI_ENTRY_UPPER_SHIFT));
+    return hGdiObj = (HGDIOBJ)(((ULONG_PTR)(hGdiObj)) | (Entry->Type << GDI_ENTRY_UPPER_SHIFT));
```

Comment:
I'm not sure the `hGdiObj =` part is really necessary since only the result of the operation is returned and nowhere else after the "return" within this function is the hGdiObj variable used yet again.